cs/turn.go

```diff
@@ -1073,9 +1073,15 @@ func (t *turn) decaySalvage() {
 func (t *turn) decayPackets() {
 	for _, packet := range t.game.MineralPackets {
 		player := t.game.getPlayer(packet.PlayerNum)
-		// update the decay rate based on this distance traveled this turn
-		decayRate := 1 - packet.getPacketDecayRate(&t.game.Rules, &player.Race)*(packet.distanceTravelled/float64(packet.WarpSpeed*packet.WarpSpeed))
-		packet.Cargo = packet.Cargo.Multiply(decayRate)
+		// update the decay amount based on this distance traveled this turn
+		decayRate := packet.getPacketDecayRate(&t.game.Rules, &player.Race)*(packet.distanceTravelled/float64(packet.WarpSpeed*packet.WarpSpeed))
```

Comment:
One more fix, if the decayRate is 0, the packet shouldn't decay at all.
```suggestion
		decayRate := packet.getPacketDecayRate(&t.game.Rules, &player.Race) * (packet.distanceTravelled / float64(packet.WarpSpeed*packet.WarpSpeed))
		if decayRate == 0 {
			continue
		}
```